src/PhpSpreadsheet/Worksheet/AutoFilter.php

```diff
@@ -453,12 +454,10 @@ private static function filterTestInPeriodDateSet($cellValue, $monthSet)
 
     /**
      * Search/Replace arrays to convert Excel wildcard syntax to a regexp syntax for preg_matching.
-     *
-     * @var array
      */
-    private static $fromReplace = ['\*', '\?', '~~', '~.*', '~.?'];
+    private const FROM_REPLACE = ['~~', '~\\*', '\\*', '~\\?', '\\?', "\x1c"];
 
-    private static $toReplace = ['.*', '.', '~', '\*', '\?'];
+    private const TO_REPLACE = ["\x1c", '[*]', '.*', '[?]', '.', '~'];
```

Comment:
> Documentation for escaping of asterisk, question mark, and tilde in text filters included spurious backslashes which are now removed.

Note that there is now a helper class for handling this, the `WildcardMatch` class that's used by the Database functions. You can find examples of its use in the `buildCondition()` method of `DatabaseAbstract` or in the `ExcelMatch` class used for the Excel `MATCH()` function.
It handles a lot more edge cases than this version: I'd forgotten that it was also needed for AutoFilter as well.

Reply:
WildcardMatch does not handle `~~` correctly. I will change it to do so, and expect to have an update to this PR shortly.

Reply:
Latest push corrects, simplifies (one str_replace in lieu of two str_replaces and a preg_replace), and comments WildcardMatch, and changes AutoFilter to use it. AutoFilter was already testing for `~~`; I added an additional test for that to MATCH as well.